### ISSUE TITLE
About MainDP the loss is NaN

### ISSUE THREAD

**Issue by zhaoyiming** [huawei-noah/GAN-pruning#15]
I have found that some parts of the mask will extremely increase and become Nan when one sample is input into the MaskBlock, although the clamp_max is set 1000.

**Comment by zhaoyiming** [comment]
BTW, Which does the pre-trained model from paper and GitHub mean? resnet56 or clipped resnet56 model.  I can not find the pre-trained model from the provided path(PyTroch-zoo and Mindspore-zoo).

**Comment by zhaoyiming** [comment]
Thank you.

**Comment by zhaoyiming** [comment]
您好，我最近在复现MainDP的一些内容，我发现Mask是在卷积计算之后和结果相乘的，这样即便一半的通道mask都为0，计算量并没有减少，现有的机器应该也不支持稀疏化特性，代码是如何做到加速呢？由于我尝试loss一直为nan，所以一直无法验证，不知道能否麻烦解答一下呢？非常感谢

**Comment by yehuitang** [comment]
Pre-trained models are required before implementing the algorithm. For ImageNet, these models can be found in the Pytorch model zoo. For CIFAR, you can train a model by yourself.  The released code is a demo about how to get the pruned model. The pruned channels will be skipped In the practical  implementation.

**Comment by zhaoyiming** [comment]
> Pre-trained models are required before implementing the algorithm. For ImageNet, these models can be found in the Pytorch model zoo. For CIFAR, you can train a model by yourself. The released code is a demo about how to get the pruned model. The pruned channels will be skipped In the practical implementation.

Thanks! "The pruned channels will be skipped In the practical implementation."  How is the pruned channel skipped?  Does this mean physically removing the channel？

**Comment by LinyeLi60** [comment]
I think ManiDP can not speed up inference without specially optimized hardware.

**Comment by zhaoyiming** [comment]
> 

Agree

**Comment by zhaoyiming** [comment]
> I think ManiDP can not speed up inference without specially optimized hardware.

Agree
